tests/integration/targets/filter_to_prettytable/tasks/main.yml

```
@@ -0,0 +1,202 @@
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Copyright (c) 2025, Timur Gadiev (tgadiev@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Install required libs
+  pip:
+    name: prettytable
+    state: present
+  delegate_to: localhost
+  become: false
+
+- name: Set test data
+  set_fact:
+    test_data:
+      - name: Alice
+        age: 25
+        role: admin
+      - name: Bob
+        age: 30
+        role: user
+    data_for_align:
+      - date: 2023-01-01
+        description: Office supplies
+        amount: 123.45
+
+# Test basic functionality
+- name: Test basic table creation
+  set_fact:
+    basic_table: '{{ test_data | community.general.to_prettytable }}'
+
+- name: Load expected basic table output
+  set_fact:
+    expected_basic_table_raw: "{{ lookup('file', 'basic_table.txt') }}"
+
+- name: Remove copyright header from expected output
+  set_fact:
+    expected_basic_table: "{{ expected_basic_table_raw.split('\n')[4:] | join('\n') }}"
```

As mentioned in a previous comment, I think it would be easier to add the resulting tables literally in here, something like:
```suggestion
    expected_basic_table: |
        +-------+-----+-------+
        |  name | age |  role |
        +-------+-----+-------+
        | Alice |  25 | admin |
        |  Bob  |  30 |  user |
        +-------+-----+-------+
```
It saves you the trouble of reading files and cleaning the licensing lines.

That being said, if you _really want_ (I see little to gain by doing that) to keep the external files, I would strongly recommend you remove the license header from them, create an adjacent file - in this example `basic_table.txt.license`, add the licensing information there so that the data file is kept clean and you don't need to remove this arbitrary number of lines from the content.

Naturally, the same argument applies to the other files as well.

---

Done